### ISSUE TITLE
Inference is too general for map of string to tuple of numbers

### ISSUE THREAD

**Issue by sbdchd** [microsoft/pyright#3314]
**Describe the bug**
Inference is too general for mapping type & indexing into the type gives an unclear type.

**To Reproduce**

```python
weekday = "Wednesday"

mapping = {
    "Sunday": (0,),
    "Monday": (1,),
    "Tuesday": (2,),
    "Wednesday": (3,),
    "Thursday": (4,),
    "Friday": (5,),
    "Saturday": (6,),
    "Weekday": (1, 2, 3, 4, 5),
    "Weekend": (0, 6),
}

day_number = mapping[weekday]

reveal_type(mapping)
reveal_type(weekday)
reveal_type(day_number)
```

Pyright says:

```
17:13 - warning: Type of "mapping" is "dict[str, Unknown]"
18:13 - warning: Type of "weekday" is "Literal['Wednesday']"
19:13 - warning: Type of "day_number" is "day_number"
```

While mypy says:

```
foo.py:17:13: note: Revealed type is "builtins.dict[builtins.str*, builtins.tuple*[builtins.int, ...]]"
foo.py:18:13: note: Revealed type is "builtins.str"
foo.py:19:13: note: Revealed type is "builtins.tuple*[builtins.int, ...]"
```

**Expected behavior**

`day_number` should be typed as `tuple[int, ...]` and `mapping` should be typed as `dict[str, tuple[int, ...]]`

I find `"day_number" is "day_number"` a little odd, shouldn't it be `"day_number" is "Unknown"`?

**VS Code extension or command-line**
Commandline: pyright 1.1.50

**Comment by JelleZijlstra** [comment]
Pyright 1.1.50 is very old, have you tried a newer version?

**Comment by sbdchd** [comment]
Oh whoops!

I wasn't calling the one installed in `node_modules`

With:

```
❯ ./node_modules/.bin/pyright --version
pyright 1.1.235
```

I get:

```
foo.py:17:13 - information: Type of "mapping" is "dict[str, Unknown]"
foo.py:18:13 - information: Type of "weekday" is "Literal['Wednesday']"
foo.py:19:13 - information: Type of "day_number" is "Unknown"
```

**Comment by JelleZijlstra** [comment]
Interesting. On latest Pylance+Pyright in VSCode I get `Type of "mapping" is "dict[str, tuple[Literal[0]] | tuple[Literal[1]] | tuple[Literal[2]] | tuple[Literal[3]] | tuple[Literal[4]] | tuple[Literal[5]] | tuple[Literal[6]] | tuple[Literal[1], Literal[2], Literal[3], Literal[4], Literal[5]] | tuple[Literal[0], Literal[6]]]"`.

**Comment by sbdchd** [comment]
Oh I needed to enable strict type checking

this config fixed it:

```
{
  "typeCheckingMode": "strict",
  "reportPrivateUsage": false,
  "reportMissingModuleSource": false,
  "reportUnnecessaryIsInstance": false
}
```
